File: zkmax/src/org/zkoss/zkmax/zul/Portalchildren.java
/* Portalchildren.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Aug 8, 2008 4:19:52 PM , Created by jumperchen
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/
package org.zkoss.zkmax.zul;

import org.zkoss.zk.ui.Component;
import org.zkoss.zk.ui.UiException;
import org.zkoss.zul.Panel;
import org.zkoss.zul.impl.XulElement;

/**
 * The column of Portallayout. <br> 
 * Child of Portalchildren can only be Panel.
 * 
 * <p>Default {@link #getZclass}: z-portal-children.
 * @author jumperchen
 * @since 3.5.0
 */
public class Portalchildren extends XulElement implements org.zkoss.zkmax.zul.api.Portalchildren {	
	public Portalchildren() {
	}

	public String getZclass() {
		return _zclass == null ? "z-portal-children" : _zclass;
	}
	
	public void beforeParentChanged(Component parent) {
		if (parent != null && !(parent instanceof Portallayout))
			throw new UiException("Wrong parent: " + parent);
		super.beforeParentChanged(parent);
	}
	
	public void beforeChildAdded(Component child, Component refChild) {
		if (!(child instanceof Panel))
			throw new UiException("Unsupported child for Portalchildren: "
					+ child);
		super.beforeChildAdded(child, refChild);
	}
}

File: zkmax/src/archive/web/js/zkmax/layout/Portalchildren.js
/* Portalchildren.js

	Purpose:
		
	Description:
		
	History:
		Thu May 27 15:17:24     2009, Created by kindalu

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under GPL Version 3.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
zkmax.layout.Portalchildren = zk.$extends(zul.Widget, {
	getZclass: function(){
		var zcls = this._zclass;
		return zcls != null ? zcls: "z-portal-children";
	},
	/*wait the panel to finish
	bind_: function() {
		this.$supers('bind_', arguments);
	},
	onChildOuter: function (child) {
		zk.on(child, "maximize", zkPortalChildren.onChildMaximize);
	},
	onChildMaximize: function (child, maximized) {
		var cave = child.parentNode, 
			cns = zk.childNodes(cave);
		for (var i = cns.length; --i >= 0;) {
			if (cns[i] != child) {
				if (maximized) {
					cns[i].style.display = "none";
				} else if (getZKAttr(cns[i], "visible") == "true") {
					cns[i].style.display = "";
					var header = $e(cns[i], "caption");
					if (header)
						header.style.cursor = "move";
					if (getZKAttr(cns[i], "maximized") == "true") {
						zkPanel.maximize(cns[i], true, true);
					} else {
						// restore the size of the panel
						zk.beforeSizeAt(cns[i]);
						zk.onSizeAt(cns[i]);
					}
				}
			}
		}
		var header = $e(child, "caption");
		if (header)
			header.style.cursor = maximized ? "default" : "move";
	},*/
	onChildAdded_: function(child){
		this.$supers('onChildAdded_', arguments);
		if(this.desktop && this.parent)
			this.parent._initDrag(child);
	},
	onChildRemoved_: function(child){
		this.$supers('onChildRemoved_', arguments);	
		if(this.desktop && this.parent)
			this.parent._cleanupDrag(child.uuid);
	}
});